### PR TITLE
tets: test mock generation works

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

the `make gen-mocks` command has been broken for a while, fixing it and adding a test to make sure it works 

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Comment by gssbzn** [comment]
example failing test without my fix 
https://github.com/mongodb/mongodb-atlas-cli/runs/5481287232?check_suite_focus=true

File: internal/latestrelease/finder.go [hunk 1]
```diff
@@ -45,14 +46,25 @@ func versionFromTag(release *github.RepositoryRelease, toolName string) string {
 	return release.GetTagName()
 }
 
+const (
+	mongoCLI = "mongocli"
+	atlasCLI = "atlascli"
+)
```
Comment (gssbzn) [inline internal/latestrelease/finder.go:52]: drive by these were being imported from the `version` package but only used here

File: internal/latestrelease/finder.go [hunk 2]
```diff
@@ -45,14 +46,25 @@ func versionFromTag(release *github.RepositoryRelease, toolName string) string {
...
 	}
 	return strings.Contains(tag, tool)
 }
 
-func (s *latestReleaseVersionFinder) searchLatestVersionPerTool(currentVersion *semver.Version, toolName string) (bool, *version.ReleaseInformation, error) {
+// ReleaseInformation Release information.
+type ReleaseInformation struct {
+	Version     string
+	PublishedAt time.Time
+}
```
Comment (gssbzn) [inline internal/latestrelease/finder.go:65]: same as above, imported from `version` but only ever used here

File: internal/version/version.go [hunk 3]
```diff
@@ -14,22 +14,8 @@
 
 package version
 
-import "time"
-
-const (
-	MongoCLI = "mongocli"
-	AtlasCLI = "atlascli"
-	owner    = "mongodb"
-)
```
Comment (gssbzn) [inline internal/version/version.go:23]: moved closer to where needed

File: internal/latestrelease/finder.go [hunk 4]
```diff
@@ -17,13 +17,14 @@ package latestrelease
...
 	"strings"
+	"time"
 
 	"github.com/Masterminds/semver/v3"
 	"github.com/google/go-github/v42/github"
 	"github.com/mongodb/mongocli/internal/version"
 )
 
-//go:generate mockgen -destination=../mocks/mock_release_version.go -package=mocks github.com/mongodb/mongocli/internal/version VersionFinder
+//go:generate mockgen -destination=../mocks/mock_release_version.go -package=mocks github.com/mongodb/mongocli/internal/latestrelease VersionFinder
```
Comment (gssbzn) [inline internal/latestrelease/finder.go:27]: fix to mock generation

File: internal/version/version.go [hunk 5]
```diff
@@ -14,22 +14,8 @@
...
 var Version string
 
 // GitCommit git sha of the build.
 var GitCommit string
-
-// ReleaseInformation Release information.
-type ReleaseInformation struct {
-	Version     string
-	PublishedAt time.Time
-}
```
Comment (gssbzn) [inline internal/version/version.go:35]: moved closer to where needed

File: .github/workflows/mocks.yml [hunk 6]
```diff
@@ -0,0 +1,32 @@
+---
```
Comment (gssbzn) [inline .github/workflows/mocks.yml:1]: inspired by the docs check

**Review by andreaangiolillo — APPROVED** [review]
LGTM 👍

**Review by fmenezes — APPROVED** [review]
LGTM good job

**Comment by gssbzn** [comment]
> good job

I had some good inspiration to base this work from :)

**Pull request merged**